smtpapi/__init__.py

```diff
@@ -34,34 +34,37 @@ def add_substitution(self, key, value):
     def set_substitutions(self, subs):
         self.data['sub'] = subs
 
+        
+    def add_attr(self, key, value, attr):
+        if attr not in self.data:
+            self.data[attr] = {}
+        self.data[attr][key] = value
+    
     def add_unique_arg(self, key, value):
-        if 'unique_args' not in self.data:
-            self.data['unique_args'] = {}
-        self.data['unique_args'][key] = value
+        add_attr(self, key, value, 'unique_args')
 
     def set_unique_args(self, value):
         self.data['unique_args'] = value
 
+    def append_attr(self, value, attr):
+        if attr not in self.data:
+            self.data[attr] = []
+        self.data[attr].append(value)
+    
     def add_category(self, category):
-        if 'category' not in self.data:
-            self.data['category'] = []
-        self.data['category'].append(category)
+        append_attr(self, category, 'category')

```

Comment:
Is there some advantage to calling the method in the above manner instead of like this?

```python
self.append_attr(category, 'category')
```